bbq/compiler/compiler_test.go

```diff
@@ -3491,13 +3491,14 @@ func TestCompileFunctionConditions(t *testing.T) {
 			[]opcode.Instruction{
 				opcode.InstructionStatement{},
 
-				// $_result = x
+				// $_result <- x
 				opcode.InstructionStatement{},
 				opcode.InstructionGetLocal{Local: xIndex},
+				opcode.InstructionTransfer{},
```

Comment:
I'm not quite sure, but it looks like this is adding an additional transfer to the program, not moving it to a different place. Given it is a resource and it is "stack-to-stack", the transfer is probably not causing a difference in slab allocations here, but is that also the case for non-resources and non-stack-to-stack transfers? 🤔

Reply:
Good point! The transfer here is correct, but it should not transfer again when returning the `$_result`. Fixed it in https://github.com/onflow/cadence/pull/4208/commits/dbe7851ae5a6427195a9414f54b08a893bceaf56